Dev

jsauer-NCAR opened the pull request:
Two sample notebooks for TTU comparisons

ewquon committed:
Update README.md

with goals outlined in meeting on Feb 22, 2019

jsauer-NCAR committed:
Merge pull request #3 from ewquon/master

Update README.md

jsauer-NCAR committed:
Merge remote-tracking branch 'upstream/master' into dev

jsauer-NCAR committed:
Merge remote-tracking branch 'upstream/dev' into dev

jsauer-NCAR committed:
Two draft notebooks, one comparing LDM's WRF results versus TTU, the second working towards TTU sonic tilt corrections

ewquon commented:
These notebooks seem reasonable to me. I'll hold off on pulling until we get the MMC_data module pulled into `mmctools` (since one of your notebooks depends on and demonstrates usage of that class), and you'll have an opportunity to clean up the notebooks (e.g., if they're telecon presentable, that's probably a good time to checkpoint your analysis)

jsauer-NCAR committed:
Merge branch 'dev' of https://github.com/a2e-mmc/assessment into dev

jsauer-NCAR committed:
Adding a workflow sequence of 3 example notebooks to 1) subsample and tilt-correct raw TTU tall-tower data with output to MMC-ASCII format, convert from MMC-ASCII format to xarray/netcdf, and finally to read in the netcdf data, make a few plots, convert from xarrays dataset to pandas ultiIndex dataframe, etc.

jsauer-NCAR committed:
Two minor fixes to filename/path building in cell 2 of each notebook.

jsauer-NCAR commented:
@ewquon Can we go ahead and merge this?

jsauer-NCAR committed:
Moved TTURawToMMC.py under assessment/SWiFT

Per Eliot's suggestions we place the TTU raw data subsampling and tilt correction helper routine under the SWiFT directory of the assessment repository.

ewquon committed:
MAJOR cleanup for consistent tab spacing

ewquon committed:
Make command-line executable

ewquon committed:
Define TTU-tower-specific parameters at module level rather than within function

ewquon committed:
Get rid of math 'import *'

ewquon committed:
Use output templates from mmctools.mmcdata

ewquon committed:
Get rid of hard-coded 10s (number of sonics)

ewquon committed:
Allow output path to be directly specified

ewquon committed:
Generalize input filenames (from DAP) and output file

Can specify outpath as a directory (in which case, a "default" filename
will be used) or as a file

ewquon committed:
Generalize input datetime

Can specify datetime string that is understandable by `to_datetime`.
Provides more control over datetime formatting (e.g., with strftime).

ewquon committed:
Add docstring

ewquon committed:
Clean up data read by using pd.read_csv()

Compute time reduced from 3min to <2min30s

At the moment, the old code has just been commented out; will remove in
future commit.

ewquon committed:
Rewrite sampling using python slicing

Results match original implementation, but sampling is inconsistent

ewquon committed:
Make resampling interval consistent

Indices with sampling stride=50 (note 0-indexed, Nt=180000 per file):
- before: [50,100,150,..,179900,179950,179999]
- after: [0,50,100,...,179900,179950]

ewquon committed:
Code clean up

ewquon committed:
Rename time variables for clarity

ewquon committed:
Remove unnecessary array declarations

These can be pulled out of the dataframe for each file

ewquon committed:
Clean up code, add more clarifying comments

ewquon committed:
Use mmctools.measurements.metmast.tilt_correction(), clean up code

ewquon committed:
More code clean up

Add vim folds for unused code

ewquon committed:
Add hashbang

ewquon committed:
Compatibility with new tilt_correction input shape (time,height)

ewquon committed:
Add TTU tower processing notebook

demonstrating tilt_correction function, using coefficients and tilt
angles from `SWiFT/TTURawToMMC.py`

ewquon committed:
Update processing/plotting notebook

jsauer-NCAR committed:
Merge pull request #1 from ewquon/feature/TTU_tilt

Add tilt correction script

jsauer-NCAR committed:
Removing old notebooks

These notebooks are now either out of date, or redundant so just ditching them altogether.
